aten/src/ATen/core/function_schema_inl.h

```diff
@@ -29,16 +29,22 @@ inline std::ostream& operator<<(std::ostream& out, const FunctionSchema& schema)
   }
 
   out << ") -> ";
-  if (schema.returns().size() == 1) {
-    out << schema.returns().at(0).type()->str();
-  } else if (schema.returns().size() > 1) {
-    out << "(";
-    for (size_t i = 0; i < schema.returns().size(); ++i) {
-      if (i > 0) out << ", ";
-      out << schema.returns()[i].type()->str();
+
+  const auto& returns = schema.returns();
+  out << "(";
```

Comment:
so the change here to align returning tuple, right? is it why all of the expect files change? are you sure it's a safe change?

I'm still getting confused by how we treat tuple return vs multiple args tbh. cc @suo @zdevito

Reply:
after some digging, I think this diff is fine with existing function schema parser. Here is the reasoning:

1) when parsing the results, we automatically remove the outer pair of parentheses. [related code](https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/script/function_schema_parser.cpp#L64) This code also tells us multiple returns should be something like `(int, float)`, one example is `aten::divmod(int x, int y) -> (int, int)` [defined here](https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/register_prim_ops.cpp#L2723)

2) since we use the outer pair parentheses to represent multi-returns, to return a tuple, we should use two pairs of parentheses. My updates on the expect files are actually correct, because the returns are all single tuples, [one example](https://github.com/pytorch/pytorch/blob/master/test/test_jit.py#L11099)

The test in https://github.com/pytorch/pytorch/pull/23208 already make sure that all the serialized schemas can be correctly parsed (i.e., parsed schemas are exactly equal to the original ones.)

Reply:
also executed the following code:
```
        s1 = parse_schema('any(Tensor self, int dim, bool keepdim=False) -> (Tensor, Tensor)')
        s2 = parse_schema('any(Tensor self, int dim, bool keepdim=False) -> ((Tensor, Tensor))')
        print(len(s1.returns))
        print(len(s2.returns))
```

output is
```
2
1
```

this also shows that, for tuple, we should use '(())'

Reply:
Yeah, existing parsing is a bit non-pythonic (as python would treat it as a tuple). However, I think we can land it avoid changing existing parser and schemas